services/src/main/java/io/druid/cli/CliHistorical.java

```diff
@@ -68,6 +69,8 @@ public CliHistorical()
           @Override
           public void configure(Binder binder)
           {
+            // register Server before binding ZkCoordinator to ensure HTTP endpoints are available immediately
+            LifecycleModule.register(binder, Server.class);
```

Comment:
When the lifecycle is started, we want the Jetty Server provided by `JettyServerModule` (and associated Jersey resources bound to it) to be started before the ServerManager lifecycle is started. binding it to the lifecycle before ServerManager is one way to achieve. Otherwise, the jetty server is not started until ServerManager is done initializing its cache, which means we can't have an HTTP endpoint report on the status of cache initialization.